src/devices/sound/nes_apu.cpp

```diff
@@ -263,53 +287,52 @@ s8 nesapu_device::apu_square(apu_t::square_t *chan)
 	while (chan->phaseacc < 0)
 	{
 		chan->phaseacc += (chan->freq >> 16);
-		chan->adder = (chan->adder + 1) & 0x0F;
+		chan->adder = (chan->adder + 1) & 0x0f;
 	}
 
 	if (chan->regs[0] & 0x10) /* fixed volume */
-		output = chan->regs[0] & 0x0F;
+		output = chan->regs[0] & 0x0f;
 	else
-		output = 0x0F - chan->env_vol;
+		output = 0x0f - chan->env_vol;
 
-	if (chan->adder < (duty_lut[chan->regs[0] >> 6]))
-		output = -output;
+	output *= BIT((duty_lut[chan->regs[0] >> 6]), 7 - BIT(chan->adder, 1, 3));
```

Comment:
This would be slightly easier to read without the extra inner parens around the duty_lut.